File: core/src/main/java/javabot/web/views/KarmaView.java
package javabot.web.views;

import com.google.inject.Injector;
import javabot.dao.KarmaDao;
import javabot.dao.util.QueryParam;
import javabot.model.Karma;

import javax.inject.Inject;
import javax.servlet.http.HttpServletRequest;
import java.util.List;

public class KarmaView extends PagedView<Karma> {
    @Inject
    private KarmaDao karmaDao;

    public KarmaView(final Injector injector, final HttpServletRequest request, final int page) {
        super(injector, request, page);
    }

    @Override
    public Long countItems() {
        return karmaDao.count();
    }

    @Override
    protected String getPageUrl() {
        return "/karma";
    }

    @Override
    public List<Karma> getPageItems() {
        return karmaDao.getKarmas(new QueryParam(getIndex(), ITEMS_PER_PAGE, "value", false));
    }

    @Override
    public String getPagedView() {
        return "karma.ftl";
    }
}

File: src/main/kotlin/javabot/web/views/KarmaView.kt
package javabot.web.views

import jakarta.inject.Inject
import jakarta.servlet.http.HttpServletRequest
import javabot.dao.AdminDao
import javabot.dao.ApiDao
import javabot.dao.ChannelDao
import javabot.dao.FactoidDao
import javabot.dao.KarmaDao
import javabot.dao.util.QueryParam
import javabot.model.Karma

class KarmaView
@Inject
constructor(
    adminDao: AdminDao,
    channelDao: ChannelDao,
    factoidDao: FactoidDao,
    apiDao: ApiDao,
    var karmaDao: KarmaDao,
    request: HttpServletRequest,
    page: Int
) : PagedView<Karma>(adminDao, channelDao, factoidDao, apiDao, request, page) {

    override fun countItems(): Long {
        return karmaDao.count()
    }

    override fun getPageUrl(): String {
        return "/karma"
    }

    override fun getPageItems(): List<Karma> {
        return karmaDao.list(QueryParam(getIndex(), ITEMS_PER_PAGE, "value", false))
    }

    override fun getPagedView(): String {
        return "karma.ftl"
    }
}
